packages/edge-gateway/src/gateway.js

```diff
@@ -115,7 +115,16 @@ export async function gatewayGet (request, env, ctx) {
     )
   }
 
-  // 3rd layer resolution - Public Gateways race
+  // 3rd layer resolution - Redirect to Public Gateway if set
+  if (env.ipfsGatewayRedirectHostname) {
+    const url = new URL(
+      `https://${cid}.ipfs.${env.ipfsGatewayRedirectHostname}${pathname}${search}`
+    )
+
+    return Response.redirect(url.toString(), 303)
```

Comment:
> [303] indicates that the redirects don't link to the requested resource itself, but to another page (such as a confirmation page, a representation of a real-world object — see [HTTP range-14](https://en.wikipedia.org/wiki/HTTPRange-14) — or an upload-progress page). This response code is often sent back as a result of [PUT](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT) or [POST](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/POST). The method used to display this redirected page is always [GET](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/GET).

Maybe we want https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/307 instead?

Reply:
@olizilla 's proposal was 303 https://hackmd.io/@olizilla/w3s-redirect . So, would like to get input if there was something that made 303 the suggestion.

For the documentation, I agree that 307 seems more appropriate

Reply:
yep, agree, 307 sounds closer to what we want. I didn't realise that "303 See other" implied we were linking to a different resource.